basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -234,6 +234,198 @@ namespace basic_autonomy
+                    if(wm->getMapRoutingGraph()->following(current_lanelet, false).empty())
+                    {
+                        throw(std::invalid_argument("No following lanelets from current lanelet reachable without a lane change, incorrectly chosen end lanelet"));
+                    }
+
+                    current_lanelet = wm->getMapRoutingGraph()->following(current_lanelet, false).front(); 
+                    auto current_lanelet_linestring = current_lanelet.centerline2d().basicLineString();    
+                    reference_centerline.insert(reference_centerline.end(), current_lanelet_linestring.begin(), current_lanelet_linestring.end());
+                    starting_lane.push_back(current_lanelet);
+                }
+            }
+
+            //Find crosstrack distance between starting lane and ending lane
+            std::vector<lanelet::BasicPoint2d> ending_lane_centerline;
+            for(size_t i = 0;i<starting_lane.size();++i){
+                lanelet::ConstLanelet curr_end_lanelet;
+                
+                if(is_lanechange_left){
+                    
+                    //get left lanelet
+                    if(wm->getMapRoutingGraph()->left(starting_lane[0])){
+                        curr_end_lanelet = wm->getMapRoutingGraph()->left(starting_lane[0]).get();
+                    }
+                    else{
+                        curr_end_lanelet = wm->getMapRoutingGraph()->adjacentLeft(starting_lane[0]).get();
+                    }
+                }
+                else{
+
+                    //get right lanelet
+                    if(wm->getMapRoutingGraph()->right(starting_lane[0])){
+                        curr_end_lanelet = wm->getMapRoutingGraph()->right(starting_lane[0]).get();
+                    }
+                    else{
+                        curr_end_lanelet = wm->getMapRoutingGraph()->adjacentRight(starting_lane[0]).get();
+                    }
+                }
+                
+                auto ending_lane_linestring = curr_end_lanelet.centerline2d().basicLineString();
+                ending_lane_centerline.insert(ending_lane_centerline.end(), ending_lane_linestring.begin(), ending_lane_linestring.end());
+                
+            }
+
+            cav_msgs::VehicleState ending_state = waypoint_generation::get_nearest_state_to_downtrack(ending_lane_centerline, ending_downtrack, wm);
+
+            int reference_line_start_index = waypoint_generation::get_nearest_point_index(reference_centerline, state);
+            int ending_line_end_index = waypoint_generation::get_nearest_point_index(ending_lane_centerline, ending_state);
+
+            lanelet::BasicLineString2d starting_centerline(reference_centerline.begin() + reference_line_start_index, reference_centerline.end());
+            lanelet::BasicLineString2d ending_centerline(ending_lane_centerline.begin(), ending_lane_centerline.begin() + ending_line_end_index);
```

Comment:
I find the mismatch of the index's between start and end centerline a bit strange. Lets says we had the lanechange crossing a lanelet boundary. The ending centerline would start from the start of the earliest lanelet on the target side then end partway through the target lanelet. While the starting centerline would start partway through the starting lanelet and go till the end of the final starting side lanelet. I'm not exactly sure if it would cause issues, but what's the advantage of doing it that way over either keeping them approximately aligned?